backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts

```diff
@@ -767,6 +769,10 @@ export class ExperimentAssignmentService {
     logger.info({ message: `Add data log userId ${userId}`, details: jsonLog });
     const keyUniqueArray: { key: string; uniquifier: string }[] = [];
 
+    if (env.mooclets.enabled) {
+      this.moocletRewardsService.parseLogsForActiveRewardMetricKey(userDoc, jsonLog, logger);
```

Comment:
Can we have a better name for this function, as this also sends the rewards?
`sendRewardsForActiveMetricsKey`